samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,145 @@
+  bool initialized_SDL   = false;
+  bool initialized_TTF   = false;
+
+  init_ret = pb_init();
+  if (init_ret != 0) {
+    debugPrint("pb_init() failed!");
+    XSleep(2000);
+    return;
+  }
+  initialized_pbkit = true;
+  pb_show_debug_screen();
+  
+  init_ret = SDL_VideoInit(NULL);
+  if (init_ret != 0) {
+    debugPrint("SDL_VideoInit failed: %s", SDL_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+  initialized_SDL = true;
+
+  init_ret = TTF_Init();
+  if (init_ret != 0) {
+    debugPrint("TTF_Init failed: %s", TTF_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+  initialized_TTF = true;
+
+  SDL_Window   *window;
+  SDL_Renderer *renderer;
+  SDL_Surface  *surface;
+  SDL_Texture  *texture;
+  TTF_Font     *font;
+
+  window = SDL_CreateWindow("NXDK SDL_ttf sample",
+                            SDL_WINDOWPOS_UNDEFINED,
+                            SDL_WINDOWPOS_UNDEFINED,
+                            SCREEN_WIDTH, SCREEN_HEIGHT,
+                            SDL_WINDOW_SHOWN);
+  if (window == NULL) {
+    debugPrint("Window creation failed: %s", SDL_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  renderer = SDL_CreateRenderer(window, -1, 0);
+  if (renderer == NULL) {
+    debugPrint("CreateRenderer failed: %s\n", SDL_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  font = TTF_OpenFont("vegur-regular.ttf", 96);
+  if (font == NULL) {
+    debugPrint("Couldn't load font: %s", TTF_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  SDL_Color textcolor = {0x7F, 0xFF, 0x7F, 0xFF};
+  surface = TTF_RenderText_Blended(font, "NXDK", textcolor);
+  TTF_CloseFont(font);
+  if (surface == NULL) {
+    debugPrint("TTF_RenderText failed: %s", TTF_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  texture = SDL_CreateTextureFromSurface(renderer, surface);
+  SDL_FreeSurface(surface);
+  if (texture == NULL) {
+    debugPrint("Couldn't create texture: %s\n", SDL_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  SDL_Rect outputPosition;
+  int window_height, window_width;
+  SDL_GetWindowSize(window, &window_width, &window_height);
+  SDL_QueryTexture(texture, NULL, NULL, &outputPosition.w, &outputPosition.h);
+  outputPosition.x = window_width/2 - outputPosition.w / 2;
+  outputPosition.y = window_height/2 - outputPosition.h / 2;
+
+  int done;
+  SDL_Event event;
+
+  /* Main render loop */
+  done = 0;
+  while (!done) {
+    /* Check for events */
+    while (SDL_PollEvent(&event)) {
+      switch (event.type) {
+      case SDL_QUIT:
+        done = 1;
+        break;
+      default:
+        break;
+      }
+    }
+    SDL_SetRenderDrawColor(renderer, 0x00, 0x00, 0x00, 0xFF);
+    SDL_RenderClear(renderer);
+    SDL_RenderCopy(renderer, texture, NULL, &outputPosition);
+    SDL_RenderPresent(renderer);
+  }
+
+  SDL_DestroyTexture(texture);
+cleanup:
+  SDL_DestroyRenderer(renderer);
```

Comment:
`renderer` might be uninitialized here.
You should probably declare all 3 of these with NULL initialization, and then check for non-NULL here.

Reply:
According to spec, SDL_DestroyRenderer and SDL_DestroyWindow only sets SDL_GetError() when an invalid pointer has been given - should be OK as we don't really care for a return value.

Reply:
To me, it still feels unclean to pass random pointers to functions. If someone was to use this code elsewhere, the uninitialized variable might point at another valid SDL renderer by sheer luck - and it would then destroy an object that was never meant to be destroyed. It's a very theoretical case, but I hope you see my point.

I'd also move the `cleanup:` one line up, so `DestroyTexture` is labeled as cleanup.

Reply:
I guess I'm narrow-minded. You make a great point. Will fix!